crates/iota-json-rpc/src/governance_api.rs

```diff
@@ -617,20 +585,65 @@ fn validator_exchange_rates(
     Ok(exchange_rates)
 }
 
-/// Check if there is any `Pending` and `Candidate` validators and get its
-/// exchange rates, this two states of a validator lifecycle can manifest
-/// during an epoch or multiple ones, is essential to not cache this data,
-/// while exchange rates for `Active` and `Inactive` validators is OK to do
-/// so because those states are manifested on epoch change.
-fn pending_and_candidate_validators_exchange_rate(
+/// Check for validators in the `Active` state and get its exchange rate
+fn active_validators_exchange_rates(
+    state: &Arc<dyn StateRead>,
+) -> RpcInterimResult<Vec<ValidatorExchangeRates>> {
+    let system_state_summary = state.get_system_state()?.into_iota_system_state_summary();
+
+    let tables = system_state_summary
+        .active_validators
+        .into_iter()
+        .map(|validator| {
+            (
+                validator.iota_address,
+                validator.staking_pool_id,
+                validator.exchange_rates_id,
+                validator.exchange_rates_size,
+                true,
+            )
+        })
+        .collect();
+
+    validator_exchange_rates(state, tables)
+}
+
+/// Check for validators in the `Inactive` state and get its exchange rate
+fn inactive_validators_exchange_rates(
+    state: &Arc<dyn StateRead>,
+) -> RpcInterimResult<Vec<ValidatorExchangeRates>> {
+    let system_state_summary = state.get_system_state()?.into_iota_system_state_summary();
+
+    let tables = validator_summary_from_system_state(
+        state,
+        system_state_summary.inactive_pools_id,
+        system_state_summary.inactive_pools_size,
+        |df| {
+            bcs::from_bytes::<ID>(&df.bcs_name).map_err(|e| {
+                IotaError::ObjectDeserialization {
+                    error: e.to_string(),
+                }
+                .into()
+            })
```

Comment:
Can be simplified through `crate::error::Error::from`

Reply:
[Fixed](https://github.com/iotaledger/iota/pull/4313/commits/2547ea52607206b7ae132cfd626c8b5783fe12da#diff-0a10b5dd25f3cf15826ae9c8e8ab585ac0e76d35d7c0f84a9334f7ed722b5406R621)